### PR TITLE
Merge challenge stage with distribution period

### PR DESCRIPTION
**Pull request by MikeHathaway**
# Description of change
## High level
* Add the 7 day challenge period into the distribution period itself. Each distribution period is now fully sequential.
  * The distribution period length is now expanded to 698400 blocks. The previous length + the challenge stage length.
  * The distribution period's end block is now the challenge stage end block. The distribution period's funding stage end block is now one block before the challenge stage starts.
  * The length of each stage are unchanged.
  * No other business logic changes.
  * Remove unnecessary clause from `startNewDistributionPeriod` checking previous periods for surplus funds to re-add to the treasury. This is now handled automatically with each distribution period automatically taking the full surplus from its predecessor.
  * Add new `testStage` unit test that rolls to each block of interest and checks the systems stage.

# Description of bug or vulnerability and solution
* [Issue non valid but recommended mitigation suggested this change](https://github.com/code-423n4/2023-05-ajna-findings/issues/387)

### CODE REVIEW

File: src/grants/base/Storage.sol [hunk 1]
```diff
@@ -27,14 +27,20 @@ abstract contract Storage is IGrantFundState {
      * @notice Length of the distribution period in blocks.
      * @dev    Roughly equivalent to the number of blocks in 90 days.
```
Comment (prat-gpt) [inline src/grants/base/Storage.sol]: The duration of distribution is now around 97 days.

File: src/grants/GrantFund.sol [hunk 2]
```diff
@@ -40,18 +40,12 @@ contract GrantFund is IGrantFund, Storage, ReentrancyGuard {
...
+        // update Treasury with unused funds from last distribution period
         {
-            // Check if any last distribution exists and its challenge stage is over
-            if (currentDistributionId > 0 && (block.number > _getChallengeStageEndBlock(currentDistributionEndBlock))) {
-                // Add unused funds from last distribution to treasury
-                _updateTreasury(currentDistributionId);
-            }
-
-            // checks if any second last distribution exist and its unused funds are not added into treasury
+            // checks if any [revious distribtuion period exists and its unused funds weren't yet re-added into the treasury
```
Comment (prat-gpt) [inline src/grants/GrantFund.sol]: nit: `[revious`

**Review by prat-gpt — APPROVED** [review]
LGTM! Just added some comments

**Review by EdNoepel — APPROVED** [review]
Aside from the typo @prateek105 noted, LGTM.

**Pull request merged**